test/mpi/errors/datatype/getcnterr.c

```diff
@@ -37,9 +37,9 @@ int main(int argc, char *argv[])
      */
     MPI_Errhandler_set(MPI_COMM_WORLD, MPI_ERRORS_RETURN);
     err = MPI_Get_elements(&recvstatus, type_ia, &count);
-    if (err == MPI_SUCCESS && count != MPI_UNDEFINED) {
+    if (err != MPI_SUCCESS && count == MPI_UNDEFINED) {
```

Comment:
According to the comment above, this call should always return an error, so I don't think this new condition is correct. The old one looks OK to me.